### ISSUE TITLE
telegraf not running on windows 

### ISSUE THREAD

**Issue by rguptarg** [influxdata/telegraf#6987]
hi team, I am getting error in windows server during telegraf service start.

WIN version and error snap are attached 
![WIN1](https://user-images.githubusercontent.com/49057178/73917887-62790e00-48e6-11ea-997d-f13927406818.JPG)

![WIN](https://user-images.githubusercontent.com/49057178/73917872-58570f80-48e6-11ea-8f2e-4b49898d5134.JPG)

**Comment by M0rdecay** [comment]
Please, try configure [agent section](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md):
`[agent]   
  quiet = false  
  debug = true
`
then, start telegraf from command line:
`telegraf.exe --config telegraf.conf`
and show output. This will help to understand the cause of the error.

**Comment by danielnelson** [comment]
I believe this may be the same issue as #6954, is this occurring after you updated Telegraf or are you just getting started?

**Comment by bdogan** [comment]
Same here at os "Windows 2012 R2 64 bit". It is core version. Manage from remote power shell. When i restart service this problem occurred. It works only after restart.

<img width="523" alt="Ekran Resmi 2020-02-10 12 01 36" src="https://user-images.githubusercontent.com/1773731/74135585-9c575680-4bfd-11ea-84cc-7a864f3ca8a2.png">
<img width="577" alt="Ekran Resmi 2020-02-10 12 00 51" src="https://user-images.githubusercontent.com/1773731/74135599-a11c0a80-4bfd-11ea-8604-eb2a32662e8c.png">
<img width="798" alt="Ekran Resmi 2020-02-10 12 00 26" src="https://user-images.githubusercontent.com/1773731/74135601-a2e5ce00-4bfd-11ea-8b68-c24fc94e3350.png">

**Comment by M0rdecay** [comment]
@bdogan, did you use agent version below 1.13? I just checked Telegraf 1.13.1 on Server 2016 and did not receive any problems with the service.

Please, show all tabs in service properties, especially "Path to executable" parameter.

**Comment by bdogan** [comment]
@M0rdecay, i have tried the following versions;

<img width="208" alt="Ekran Resmi 2020-02-10 13 06 23" src="https://user-images.githubusercontent.com/1773731/74141075-530c0480-4c07-11ea-8da5-b2a696a1fa38.png">



All tabs;

<img width="431" alt="Ekran Resmi 2020-02-10 13 06 43" src="https://user-images.githubusercontent.com/1773731/74141408-eb09ee00-4c07-11ea-9c67-7578dd2c0a67.png">
<img width="429" alt="Ekran Resmi 2020-02-10 13 06 51" src="https://user-images.githubusercontent.com/1773731/74141416-ee9d7500-4c07-11ea-9ba6-a2fa2dd860e1.png">
<img width="430" alt="Ekran Resmi 2020-02-10 13 06 55" src="https://user-images.githubusercontent.com/1773731/74141421-ef360b80-4c07-11ea-815a-5996c73e23e4.png">
<img width="431" alt="Ekran Resmi 2020-02-10 13 06 59" src="https://user-images.githubusercontent.com/1773731/74141426-f0673880-4c07-11ea-91ee-e781e890340b.png">

Path to Executable;
```"C:\Program Files\Telegraf\telegraf.exe" --config "C:\Program Files\Telegraf\telegraf.conf" --service-name telegraf```

**Comment by M0rdecay** [comment]
@bdogan, thanks for the info, everything seems to be fine. If you are not tired, let's try another option. Record the full trace with [procmon](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon) during service start.

**Comment by bdogan** [comment]
Here is the trace file [telegraf_trace.zip](https://github.com/influxdata/telegraf/files/4180630/telegraf_trace.zip) (PML format).

Note: I have filtered only ***process name is not "w3wp.exe"***

**Comment by M0rdecay** [comment]
Thanks for the trace. This is strange, but no errors in the operation of telegraf or services.exe were noticed in it. It can be seen that the agent is launched on the initiative services.exe and works for a minute. 
And, since you encounter this error when using any version and architecture of the agent, I will assume that the service cannot be authorized under the local system account.

Perhaps this is still connected with the resource \\NAS1\IPC$, since an attempt to access it is the last meaningful recorded action of the agent.
The penultimate such action is an attempt to find the chronyc executable, but only Telegraf developers know what role it plays. Telegraf cannot find the file.

If you can deal with your problem, please describe the solution.

**Comment by danielnelson** [comment]
> The penultimate such action is an attempt to find the chronyc executable, but only Telegraf developers know what role it plays. Telegraf cannot find the file.

Looks like we search for `chronyc` when registering the `chrony` plugin at startup, I suspect it is unrelated to this problem but I'll move it so that we only do this if the plugin is enabled.

**Comment by AlexHeylin** [comment]
I've found a similar issue has occurred on one of my servers (Win 2012 R2), where the service failed to respond / start following each of several OS reboots (leading to no metrics for 36 hours).  However once OS has been up a while, service starts without issue.  The server should not be heavily loaded during / after boot, and telegraf is light so there should be no reason for it to be unresponsive. 

Unfortunately in trying to debug this it seems the log handling on Windows is probably not as intended and you can't run debug to Eventlog (in addition to stderr always reporting eventlog is not valid option, due to missing case in switch at logger.go line 106).  Also it seems to only output debug for the output, not the inputs or main code - so it's really not very helpful for diags. 

I'm happy to do some digging on my server if it helps?

**Comment by powersj** [comment]
Hi,

This is an older bug and we have not had any additional reports, plus we have made some more enhancements to how the service runs on Windows. I am going to go ahead and close this issue. 

If you are still running into issues please feel free to file a new ticket with logs and configs!

Thanks!
